src/biogeochem/CNCStateUpdate1Mod.F90

```diff
@@ -69,6 +69,14 @@ subroutine CStateUpdateDynPatch(bounds, num_soilc_with_inactive, filter_soilc_wi
        do j = 1,nlevdecomp
           do fc = 1, num_soilc_with_inactive
              c = filter_soilc_with_inactive(fc)
+             ! TODO BFB refactor: In loops like this, we will run into trouble
+             !      with MIMICS bc we will have 2 instead of 3 litter pools. Can
+             !      address with if statements (NOT ideal) or
+             !      turn these repetitive lines into a loop BUT to do that
+             !      we need to turn dwt_frootc_to_litr_* into a 3d var.
+             !      There are 3 such loops here, 2 in CNCStateUpdate2, and one
+             !      in CNCStateUpdate1 and the pattern repeats in the
+             !      corresponding N modules.
```

Comment:
@wwieder and @ekluzek I'm interested in your input regarding this refactor. It seems straightforward but it touches a fair number of variables. One way or another we will have to address this for MIMICS to work, correct me if I'm wrong.

Reply:
I posted an update of this in [the refactor PR.](https://github.com/ESCOMP/CTSM/pull/1340/files#r613643764)